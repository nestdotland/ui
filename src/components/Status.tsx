import React from "react";
import { Tag, TagLabel } from "@chakra-ui/react";

export interface StatusProps {
  status: "normal" | "degraded" | "partial" | "down";
}

/**
 * Add Custom Components
 */
export const Status = (props: StatusProps) => {
  const status = props.status ?? "normal";

  const labels = {
    normal: "All Systems Normal",
    degraded: "Degraded Performance",
    partial: "Partial Outage",
    down: "Complete Outage",
  };

  const colors = {
    normal: "green",
    degraded: "yellow",
    partial: "orange",
    down: "red",
  };

  return (
    <Tag colorScheme={colors[status]} {...props}>
      <TagLabel>â¬¤ {labels[status]}</TagLabel>
    </Tag>
  );
};
