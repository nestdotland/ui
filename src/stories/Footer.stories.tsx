import React from "react";
import { Meta, Story } from "@storybook/react";
import { Footer, FooterProps } from "../";

const meta: Meta = {
  title: "Footer",
  component: Footer,
  parameters: {
    controls: { expanded: true },
  },
};

export default meta;

const Template: Story<FooterProps> = (args) => <Footer {...args} />;

export const Default = Template.bind({});

Default.args = {};
