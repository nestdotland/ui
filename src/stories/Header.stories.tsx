import React from "react";
import { Meta, Story } from "@storybook/react";
import { Text, Heading } from "@chakra-ui/react";
import { Header, HeaderProps } from "../";

const meta: Meta = {
  title: "Header",
  component: Header,
  parameters: {
    controls: { expanded: true },
  },
};

export default meta;

const Template: Story<HeaderProps> = (args) => {
  return (
    <Header alignItems="center" {...args}>
      <Heading as="h1" size="xl" color="primary">
        Nest
      </Heading>
      <Text as="h2" fontSize="2xl" py="8" align="center">
        An{" "}
        <Text as="span" fontWeight="bold">
          immutable
        </Text>{" "}
        module registry for Deno.
      </Text>
    </Header>
  );
};

export const Default = Template.bind({});

Default.args = {};
