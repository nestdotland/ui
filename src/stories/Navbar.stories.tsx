import React from "react";
import { Meta, Story } from "@storybook/react";
import { Navbar, NavbarProps } from "..";
import { SiGithub, SiDiscord } from "react-icons/si";

const config = {
  pages: [
    { name: "Modules", link: "#modules" },
    { name: "Blog", link: "#blog" },
    { name: "Docs", link: "#docs" },
    { name: "Login", link: "#login", active: true },
  ],
  socials: [
    { name: "GitHub", link: "#github", icon: SiGithub },
    { name: "Discord", link: "#discord", icon: SiDiscord },
  ],
};

const meta: Meta = {
  title: "Navbar",
  component: Navbar,
  parameters: {
    controls: { expanded: true },
  },
};

export default meta;

const Template: Story<NavbarProps> = (args) => <Navbar {...args} />;

export const Default = Template.bind({});

Default.args = { ...config };
