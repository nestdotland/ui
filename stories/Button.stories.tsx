import React from "react";
import { Meta, Story } from "@storybook/react";
import { Button, ButtonProps } from "@chakra-ui/react";

const meta: Meta = {
  title: "Button",
  component: Button,
  argTypes: {
    children: {
      control: {
        type: "text",
      },
    },
    colorScheme: {
      control: {
        type: "text",
      },
    },
  },
  parameters: {
    controls: { expanded: true },
  },
};

export default meta;

const Template: Story<ButtonProps> = (args) => {
  return <Button {...args} />;
};

export const Default = Template.bind({});

Default.args = {
  children: "Button",
  colorScheme: "primary",
};
