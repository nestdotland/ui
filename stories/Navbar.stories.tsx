import React from "react";
import { Meta, Story } from "@storybook/react";
import { Navbar, NavbarProps } from "../src";
import { SiGithub, SiDiscord } from "react-icons/si";

const config = {
  pages: [
    { name: "Modules", link: "#" },
    { name: "Blog", link: "#" },
    { name: "Docs", link: "#" },
  ],
  socials: [
    { name: "GitHub", link: "#", icon: SiGithub },
    { name: "Discord", link: "#", icon: SiDiscord },
  ],
};

const meta: Meta = {
  title: "Navbar",
  component: Navbar,
  parameters: {
    controls: { expanded: true },
  },
};

export default meta;

const Template: Story<NavbarProps> = (args) => <Navbar {...config} {...args} />;

// By passing using the Args format for exported stories, you can control the props for a component for reuse in a test
// https://storybook.js.org/docs/react/workflows/unit-testing
export const Default = Template.bind({});

Default.args = {};
