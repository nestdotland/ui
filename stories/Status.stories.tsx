import React from "react";
import { Meta, Story } from "@storybook/react";
import { Status, StatusProps } from "../src";

const meta: Meta = {
  title: "Status",
  component: Status,

  args: {
    status: "normal",
  },
};

export default meta;

const Template: Story<StatusProps> = (args) => (
  <Status status="normal" {...args} />
);

export const Default = Template.bind({});

Default.args = {};
