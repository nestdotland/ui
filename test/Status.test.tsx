import React from "react";
import * as ReactDOM from "react-dom";
import { Default as Status } from "../stories/Status.stories";

describe("Status", () => {
  it("renders without failing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Status status="normal" />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
});
