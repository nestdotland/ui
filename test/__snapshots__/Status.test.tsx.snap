// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status renders without failing 1`] = `
<bound Template
  status="normal"
/>
`;
